cookbooks/continental_extension.prm

```diff
@@ -0,0 +1,256 @@
+    set Function expression = if( (h-y)<=20.e3, \
+                                  ts1 + (qs1/k1)*(h-y) - (A*(h-y)*(h-y))/(2.0*k1), \
+                                  if((h-y)>20.e3 && (h-y)<=30.e3, ts2 + (qs2/k2)*(h-y-20.e3),\
+                                  ts3 + (qs3/k3)*(h-y-30.e3)));
+  end
+end
+
+# Internal heating (only internal heating for crust)
+# Assuming a very large decay time, the heat production (W/m^3) is effectively:
+#   radioactive_heating_rate (W/kg) * density (kg/m^3)
+# For a reference crustal density of 2800 kg/m^3 and 1.5e-6 W/m^3 heating production,
+# the heating rate (assuming a single element) is 1.5e-6/2800. = 5.357142857e-10.
+subsection Heating model
+  set Model name = radioactive decay
+  subsection Radioactive decay
+    set Number of elements    = 1
+    set Heating rates                 = 5.357142857e-10
+    set Half decay times              = 1.e20
+    set Initial concentrations mantle = 0.0
+    set Initial concentrations crust  = 1.0
+    set Crust defined by composition  = true
+    set Crust composition number      = 0
+  end
+end
+
+# Material model
+# Rheology: Non-linear viscous flow and Drucker Prager Plasticity
+# Values for most rheological parameters are specified for a background material and
+# each compositional field.  Values for viscous deformation are based on dislocation
+# creep flow-laws, with distinct values for the upper crust (wet quartzite), lower
+# crust (wet anorthite) and mantle (dry olivine).  Table 1 of Naliboff and Buiter (2015),
+# Earth Planet. Sci. Lett., v.421, p. 58-67 contains values for each of these flow laws.     
+subsection Material model
+  set Model name = visco plastic
+
+  subsection Visco Plastic
+
+    # Reference temperature and viscosity
+    set Reference temperature = 293
+    set Reference viscosity = 1e22
+
+    # As the boundaries are all free-slip, so iUse the reference strain-rate
+    # (used on first nonlinear iteration) to specify the strain-rate used
+    # in the viscous flow law.  Set the minimum strain-rate to the same value.
+
+    # The minimum strain-rate helps limit large viscosities values that arise
+    # as the strain-rate approaches zero.
+    # The reference strain-rate is used on the first non-linear iteration
+    # of the first time step when the velocity has not been determined yet. 
+    set Minimum strain rate = 1.e-20
+    set Reference strain rate = 1.e-16
+
+    # Limit the viscosity with minimum and maximum values
+    set Minimum viscosity = 1e18
+    set Maximum viscosity = 1e26
+
+    # Thermal diffusivity is adjusted to match thermal conductivities
+    # assumed in the assigning the initial geotherm
+    set Thermal diffusivities = 1.333333e-6,1.190476e-6,1.149425e-6,1.333333e-6,1.333333e-6
+    set Heat capacities = 750.,750.,750.,750.,750.
+    set Densities = 3300,2800,2900,3300,3300
+    set Thermal expansivities = 2e-5,2e-5,2e-5,2e-5,2e-5
```

Comment:
Do you think it would be easier to read these values if the equals signs were aligned vertically, and also the commas that separate numbers?